codegen/cmd/injection-gen/generators/reconciler_controller.go

```diff
@@ -227,10 +239,17 @@ func NewImpl(ctx {{.contextContext|raw}}, r Interface{{if .hasClass}}, classValu
 		{{if .hasClass}}classValue: classValue,{{end}}
 	}
 
-	t := {{.reflectTypeOf|raw}}(r).Elem()
-	queueName := {{.fmtSprintf|raw}}("%s.%s", {{.stringsReplaceAll|raw}}(t.PkgPath(), "/", "-"), t.Name())
+	ctrType := {{.reflectTypeOf|raw}}(r).Elem()
+	ctrTypeName := {{.fmtSprintf|raw}}("%s.%s", ctrType.PkgPath(), ctrType.Name())
+	ctrTypeName = {{.stringsReplaceAll|raw}}(ctrTypeName, "/", ".")
+
+	logger = logger.With(
+			{{.zapString|raw}}({{.logkeyControllerType|raw}}, ctrTypeName),
+			{{.zapString|raw}}({{.logkeyControllerKind|raw}}, "{{ printf "%s.%s" .group .type.Name.Name }}"),
+	)
```

Comment:
(very) Late comment, but this change has a side effect of creating new leases with a different name.

Because it wasn't mentioned anywhere, here are the actions I had to take, in case someone else runs into issues after upgrading:

* RBAC rules that allow changes to specific lease names must be updated
* old leases can be removed from clusters

Reply:
Not in serving core, but downstream potentially has RBAC rules that further restrict the list of leases which controllers are allowed to modify (versus "everything in that namespace").

That was the case in the GitLab event source, for example (eventing). We at TriggerMesh also apply such restrictions.

Reply:
Is there a reason for not creating new namespaces for tenancy vs. hardcoding lease names in RBAC rules?

You're effectively limiting the controller from being HA